### PR TITLE
Add BDK example

### PR DESCRIPTION
**Pull request by klochowicz**
Dart VM is calling into sync Rust without blocking the Flutter thread.

### CODE REVIEW

**Comment by klochowicz** [comment]
@holzeis I've rebased it mostly to see how much cleaner it would look after removing the generated code from git, feel free to merge it or just use it as a part of your work going forward; there's not too much raw value in this one :)

**Comment by holzeis** [comment]
@klochowicz much better 👍

File: rust/src/api.rs [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
+use std::time::Duration;
+
+use crate::wallet;
 use anyhow::anyhow;
 use anyhow::Result;
-
 use flutter_rust_bridge::ZeroCopyBuffer;
 
+pub fn init_wallet() -> Result<String> {
+    println!("going to sleep");
```
Comment (bonomat) [inline rust/src/api.rs:10]: Do we have a logger already?

Comment (bonomat) [reply rust/src/api.rs:10]: I guess this ticket will cover it: https://github.com/itchysats/10101/issues/83

Comment (klochowicz) [reply rust/src/api.rs:10]: no logger yet! we will disable `println` etc via clippy lint soon when we will add one.

**Review by da-kami — COMMENTED** [review]
One sentence with a bit more context would have helped.

My understanding: 
This PR is to test that we can call `sync` on the wallet and we see an updated balance (?)

Are you planning to merge this, or is this meant to be as a Draft PR that showcases how this will work? :)
If this is just a showcase maybe we can mark it as `Draft` PR until it is something that is mergable.

**Comment by da-kami** [comment]
Earlier we talked about this work, and you mentioned that you want to drive this further by returning the wallet from the `init` function and then the wallet's lifecycle would be in control of dart and we would just call functions on the `wallet` object (struct) from dart. 

Are you planning to add this to this PR?

**Comment by klochowicz** [comment]
> Earlier we talked about this work, and you mentioned that you want to drive this further by returning the wallet from the `init` function and then the wallet's lifecycle would be in control of dart and we would just call functions on the `wallet` object (struct) from dart.



> One sentence with a bit more context would have helped.
> 
> My understanding: This PR is to test that we can call `sync` on the wallet and we see an updated balance (?)
> 
> Are you planning to merge this, or is this meant to be as a Draft PR that showcases how this will work? :) If this is just a showcase maybe we can mark it as `Draft` PR until it is something that is mergable.

I didn't set it out as draft, as I was planning on merging this, so @holzeis could  build on top of it last Thursday. 
If you don't think it's mergeable, I'm happy to close it.

**Comment by klochowicz** [comment]
> Earlier we talked about this work, and you mentioned that you want to drive this further by returning the wallet from the `init` function and then the wallet's lifecycle would be in control of dart and we would just call functions on the `wallet` object (struct) from dart.
> 
> Are you planning to add this to this PR?

I planned to add state management as a separate PR, because otherwise I'm needlessly blocking others. Can rework this one if you prefer this way.

**Comment by klochowicz** [comment]
> One sentence with a bit more context would have helped.

thanks for the feedback! will try to be more explicit next time :)

**Comment by klochowicz** [comment]
this PR has served its purpose, and there's not much usefulness in it anymore.

**Pull request closed without merging**